Tests/TestCesiumOriginShift.cs

```diff
@@ -97,4 +97,96 @@ public IEnumerator UsesGeoreferenceTransformForActivationCheck()
+
+        CesiumOriginShift originShift = goOriginShift.AddComponent<CesiumOriginShift>();
+        originShift.useActivationDistance = true;
+        originShift.activationDistance = 5000;
+
+        yield return null;
+
+        globeAnchor.positionGlobeFixed = baseEcef + new double3(4999.0, 0, 0);
+
+        yield return null;
+
+        Assert.AreEqual(baseEcef.x + 4999.0, globeAnchor.positionGlobeFixed.x);
+        Assert.AreEqual(baseEcef.x + 0, georeference.ecefX);
+
+        globeAnchor.positionGlobeFixed = baseEcef + new double3(5010.0, 0, 0);
+        yield return null;
+
+        Assert.AreEqual(baseEcef.x + 5010.0, globeAnchor.positionGlobeFixed.x);
+        Assert.AreEqual(baseEcef.x + 5010.0, georeference.ecefX);
+    }
+
+    // Testing a bug where the character controller would cause a jump ahead when the activation distance is hit.
+    [UnityTest]
+    public IEnumerator ShiftingOriginWithCharacterController()
+    {
+        double3 baseEcef = new double3(-2694020, -4297355, 3854720);
+
+        GameObject goGeoreference = new GameObject("Georeference");
+        CesiumGeoreference georeference = goGeoreference.AddComponent<CesiumGeoreference>();
+        georeference.ecefX = baseEcef.x;
+        georeference.ecefY = baseEcef.y;
+        georeference.ecefZ = baseEcef.z;
+
+        GameObject goOriginShift = new GameObject("OriginShifter");
+        goOriginShift.transform.parent = goGeoreference.transform;
+        CesiumGlobeAnchor globeAnchor = goOriginShift.AddComponent<CesiumGlobeAnchor>();
+
+        CesiumOriginShift originShift = goOriginShift.AddComponent<CesiumOriginShift>();
+        originShift.useActivationDistance = true;
+        originShift.activationDistance = 5000;
+
+        CharacterController controller = goOriginShift.AddComponent<CharacterController>();
+        controller.radius = 1.0f;
+        controller.height = 1.0f;
+        controller.center = Vector3.zero;
+        controller.enableOverlapRecovery = false;
+
+        yield return new WaitForEndOfFrame();
+
+        Assert.AreEqual((float)baseEcef.x, (float)globeAnchor.positionGlobeFixed.x);
+
+        // speed per second
+        double speed = 1000.0;
+        double duration = 10.0;
+        float startTime = Time.time;
+        Vector3 startPos = globeAnchor.transform.position;
+
+        while ((Time.time - startTime) < duration)
+        {
+            double3 previousPositionEcef = globeAnchor.positionGlobeFixed.x;
+
+            yield return new WaitForFixedUpdate();
+
+            double unitsEcef = speed * Time.deltaTime;
+            double3 movement = georeference.TransformEarthCenteredEarthFixedDirectionToUnity(new double3(unitsEcef, 0, 0));
+            controller.Move((float3)movement);
+
+            yield return new WaitForEndOfFrame();
+
+            globeAnchor.Sync();
+            Assert.AreEqual((float)(previousPositionEcef.x + unitsEcef), (float)globeAnchor.positionGlobeFixed.x);
```

Comment:
For these float-arithmetic checks, can we use epsilon equality just to be safe? Similar to this test in `TestCesiumGeoreference`

https://github.com/CesiumGS/cesium-unity/blob/db4df38750eae7e5ed8be4b9669d2171c8345f80/Tests/TestCesiumGeoreference.cs#L71